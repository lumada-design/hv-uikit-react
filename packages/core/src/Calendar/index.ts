export * from "./Calendar";
export * from "./CalendarHeader";
export * from "./types";
