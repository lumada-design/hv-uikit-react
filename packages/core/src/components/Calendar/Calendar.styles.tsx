import styled from "@emotion/styled";
import { theme } from "@hitachivantara/uikit-styles";
import calendarClasses from "./calendarClasses";
import { HvSingleCalendar, HvSingleCalendarProps } from "./SingleCalendar";

export const StyledRoot = styled("div")({
  display: "flex",
  justifyContent: "center",
});

export const StyledRangeCalendarContainer = styled("div")({
  display: "flex",
});

export const StyledSingleCalendar = styled((props: HvSingleCalendarProps) => (
  <HvSingleCalendar {...props} />
))({
  "&:hover": {
    [`& ~ .${calendarClasses.singleCalendar}`]: {
      // target the next calendar when the former is hovered
      "& > div": {
        "& > div:nth-child(3)": {
          "& > div": {
            "& > div[data-calendar-cell='calendarCell']": {
              backgroundColor: theme.colors.atmo1,
            },
          },
        },
      },
    },
  },
});
