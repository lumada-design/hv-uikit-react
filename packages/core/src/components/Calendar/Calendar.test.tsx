import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { HvCalendar } from "./Calendar";

describe("<Calendar /> with minimum configuration", () => {
  it("should render composed navigation components", () => {
    const { container } = render(<HvCalendar />);

    expect(
      container.getElementsByClassName(
        "HvComposedNavigation-navigationContainer"
      )
    ).toHaveLength(1);
  });
});

describe("<Calendar /> with configurations", () => {
  it("renders the month and year", () => {
    const selectedDate = new Date("2020-01-10");
    render(<HvCalendar value={selectedDate} />);

    expect(screen.getByText("January")).toBeInTheDocument();
    expect(screen.getByText("2023")).toBeInTheDocument();
  });
});
