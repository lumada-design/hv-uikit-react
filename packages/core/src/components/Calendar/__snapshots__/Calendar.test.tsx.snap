// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Calendar /> with configurations > should render correctly 1`] = `
<div>
  <div
    class="HvCalendar-root css-xnsj8j-StyledRoot e6gej3i2"
  >
    <div
      class="HvSingleCalendar-calendarContainer css-1e2uj5c-StyledCalendarContainer e10ihr3e2"
    >
      <div
        class="HvSingleCalendar-calendarWrapper css-1otz5vv-StyledCalendarWrapper e10ihr3e1"
        id="default"
      >
        <div
          class="HvCalendarHeader-root css-82wps1-StyledRoot e1yjj6ap4"
          id="default-header"
        >
          <div
            class="HvCalendarHeader-headerDate"
          >
            <input
              class="HvCalendarHeader-input css-fhzinf-StyledInput e1yjj6ap2"
              id="default-header-header-input"
              placeholder="MM/DD/YYYY"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          style="height: 32px;"
        />
        <div>
          <div
            class="HvComposedNavigation-navigationContainer css-1rxbfsa-StyledNavigationContainer es2ikxx1"
          >
            <div
              class="HvComposedNavigation-navigationMonth css-11pr0v9-StyledNavigationMonth es2ikxx0"
            >
              <div
                class="HvNavigation-root css-naomxv-StyledRoot e1ozyvih3"
              >
                <div
                  class="HvNavigation-icon e1ozyvih2 css-1nvi80g-StyledDropLeftIcon"
                  id="default-navigation-month-left"
                  name="DropLeftXS"
                  tabindex="0"
                >
                  <svg
                    focusable="false"
                    height="12"
                    viewBox="0 0 12 12"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      class="color0"
                      d="m3.35 5.99994 4.6-4.6.7.7L4.75 6l3.9 3.9-.7.7z"
                      fill="var(--uikit-colors-secondary)"
                    />
                  </svg>
                </div>
                <div
                  class="HvNavigation-text css-q82srd-StyledText e1ozyvih0"
                  id="default-navigation-month"
                  role="presentation"
                  tabindex="0"
                >
                  <p
                    class="HvTypography-root HvTypography-body hv-1gxwolr"
                  >
                    January
                  </p>
                </div>
                <div
                  class="HvNavigation-icon  e1ozyvih1 css-16jdwg-StyledDropRightIcon"
                  id="default-navigation-month-right"
                  name="DropRightXS"
                  tabindex="0"
                >
                  <svg
                    focusable="false"
                    height="12"
                    viewBox="0 0 12 12"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      class="color0"
                      d="m4.05 10.6-.7-.7L7.25 6l-3.9-3.9.7-.7L8.65 6z"
                      fill="var(--uikit-colors-secondary)"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="HvNavigation-root css-naomxv-StyledRoot e1ozyvih3"
            >
              <div
                class="HvNavigation-icon e1ozyvih2 css-1nvi80g-StyledDropLeftIcon"
                id="default-navigation-year-left"
                name="DropLeftXS"
                tabindex="0"
              >
                <svg
                  focusable="false"
                  height="12"
                  viewBox="0 0 12 12"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="color0"
                    d="m3.35 5.99994 4.6-4.6.7.7L4.75 6l3.9 3.9-.7.7z"
                    fill="var(--uikit-colors-secondary)"
                  />
                </svg>
              </div>
              <div
                class="HvNavigation-textWithoutHover css-q82srd-StyledText e1ozyvih0"
                id="default-navigation-year"
                role="presentation"
                tabindex="-1"
              >
                <p
                  class="HvTypography-root HvTypography-body hv-1gxwolr"
                >
                  2023
                </p>
              </div>
              <div
                class="HvNavigation-icon  e1ozyvih1 css-16jdwg-StyledDropRightIcon"
                id="default-navigation-year-right"
                name="DropRightXS"
                tabindex="0"
              >
                <svg
                  focusable="false"
                  height="12"
                  viewBox="0 0 12 12"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="color0"
                    d="m4.05 10.6-.7-.7L7.25 6l-3.9-3.9.7-.7L8.65 6z"
                    fill="var(--uikit-colors-secondary)"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="HvSingleCalendar-calendarGrid css-1v8f9lt-StyledCalendarGrid e10ihr3e0"
          >
             
            <label
              class="HvTypography-root HvTypography-label HvCalendarWeekLabels-calendarDay css-13lkucx-StyledCalendarDay exstlpd0 hv-1j8ohlw"
            >
              S
            </label>
            <label
              class="HvTypography-root HvTypography-label HvCalendarWeekLabels-calendarDay css-13lkucx-StyledCalendarDay exstlpd0 hv-1j8ohlw"
            >
              M
            </label>
            <label
              class="HvTypography-root HvTypography-label HvCalendarWeekLabels-calendarDay css-13lkucx-StyledCalendarDay exstlpd0 hv-1j8ohlw"
            >
              T
            </label>
            <label
              class="HvTypography-root HvTypography-label HvCalendarWeekLabels-calendarDay css-13lkucx-StyledCalendarDay exstlpd0 hv-1j8ohlw"
            >
              W
            </label>
            <label
              class="HvTypography-root HvTypography-label HvCalendarWeekLabels-calendarDay css-13lkucx-StyledCalendarDay exstlpd0 hv-1j8ohlw"
            >
              T
            </label>
            <label
              class="HvTypography-root HvTypography-label HvCalendarWeekLabels-calendarDay css-13lkucx-StyledCalendarDay exstlpd0 hv-1j8ohlw"
            >
              F
            </label>
            <label
              class="HvTypography-root HvTypography-label HvCalendarWeekLabels-calendarDay css-13lkucx-StyledCalendarDay exstlpd0 hv-1j8ohlw"
            >
              S
            </label>
             
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  28
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  29
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  30
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  31
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="0"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateSelected css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  1
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  2
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  3
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  4
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  5
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  6
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  7
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  8
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  9
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  10
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  11
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  12
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  13
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  14
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  15
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  16
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  17
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  18
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  19
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  20
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  21
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  22
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  23
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  24
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  25
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  26
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  27
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  28
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  29
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  30
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer HvSingleCalendar-focusSelection css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="true"
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  31
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  1
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  2
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  3
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  4
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  5
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  6
                </p>
              </button>
            </div>
            <div
              class="HvSingleCalendar-dateWrapper css-glu5si-StyledDateWrapper eulxq3n0"
              data-calendar-cell="calendarCell"
            >
              <button
                class="HvSingleCalendar-cellContainer css-1unh1h1-StyledCellContainer eulxq3n2"
                data-in-month="false"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <p
                  class="HvTypography-root HvTypography-body HvSingleCalendar-calendarDate HvSingleCalendar-calendarDateNotInMonth css-1sl39ae-StyledCalendarDate eulxq3n1 hv-1gxwolr"
                >
                  7
                </p>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
